bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * The {@link FreeAtHomeBridgeHandler} is responsible for handling the free@home bridge and
+ * its main communication.
+ *
+ * @author Andras Uhrin - Initial contribution
+ * @param <FreeAtHomeSystemHttpQueueResponse>
+ *
+ */
+@NonNullByDefault
+public class FreeAtHomeBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeBridgeHandler.class);
+
+    public @Nullable static FreeAtHomeBridgeHandler freeAtHomeSystemHandler = null;
+
+    public ChannelUpdateHandler channelUpdateHandler = new ChannelUpdateHandler();
+
+    // Clients for the network communication
+    private HttpClient httpClient;
+    private @Nullable WebSocketClient websocketClient = null;
+    private @Nullable EventSocket socket = null;
+    private @Nullable FreeAtHomeWebsocketMonitorThread socketMonitor = null;
+    private @Nullable QueuedThreadPool jettyThreadPool = null;
+
+    private String sysApUID = "00000000-0000-0000-0000-000000000000";
+    private String ipAddress = "192.168.1.1";
+    private String username = "";
+    private String password = "";
+
+    private String baseUrl = "";
+
+    private String authField = "";
+
+    private AtomicBoolean httpConnectionOK = new AtomicBoolean(false);
+
+    int numberOfComponents = 0;
+
+    private static final int BRIDGE_WEBSOCKET_RECONNECT_DELAY = 60;
+
+    private List<String> listOfComponentId = new ArrayList<String>();
+
+    public FreeAtHomeBridgeHandler(Bridge thing, HttpClient client) {
+        super(thing);
+
+        httpClient = client;
+    }
+
+    /**
+     * stub method for handlCommand
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.warn("Unknown handle command for the bridge - channellUID {}, command {}", channelUID, command);
+    }
+
+    /**
+     * Method to get the device list
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @SuppressWarnings("deprecation")
+    public @Nullable List<String> getDeviceDeviceList() {
+        boolean ret = false;
+
+        listOfComponentId.clear();
+
+        String url = baseUrl + "/rest/devicelist";
+
+        // Perform a simple GET and wait for the response.
+        try {
+            HttpClient client = httpClient;
+
+            if (client != null) {
```

Comment:
`httpClient` is provided by the constructor, how can it be null?

Reply:
The constructor is getting the httpClient from the HandlerFactory. If the httpClient provided to the HandlerFactory could be never null. This check is obsolate.

Is it true, that the HandlerFactor will get every time a valid httpClient?

Reply:
It will get a valid `HttpClientFactory`, which will provide a valid HttpClient when calling `getCommonHttpClient()`.

Reply:
null check is removed